packages/create-block/lib/templates.js

```diff
@@ -1,91 +1,89 @@
+/**
+ * External dependencies
+ */
+const glob = require( 'fast-glob' );
+const { readFile } = require( 'fs' ).promises;
+const { fromPairs } = require( 'lodash' );
+const { join } = require( 'path' );
+
 /**
  * Internal dependencies
  */
 const CLIError = require( './cli-error' );
 const prompts = require( './prompts' );
 
-const namespace = 'create-block';
-const dashicon = 'smiley';
-const category = 'widgets';
-const author = 'The WordPress Contributors';
-const license = 'GPL-2.0-or-later';
-const licenseURI = 'https://www.gnu.org/licenses/gpl-2.0.html';
-const version = '0.1.0';
-
-const templates = {
+const predefinedBlockTemplates = {
 	es5: {
 		defaultValues: {
-			namespace,
 			slug: 'es5-example',
 			title: 'ES5 Example',
 			description:
 				'Example block written with ES5 standard and no JSX – no build step required.',
-			dashicon,
-			category,
-			author,
-			license,
-			licenseURI,
-			version,
 		},
-		outputFiles: [
-			'.editorconfig',
-			'editor.css',
-			'index.js',
-			'$slug.php',
-			'style.css',
-			'readme.txt',
-		],
+		wpScripts: false,
 	},
 	esnext: {
 		defaultValues: {
-			namespace,
 			slug: 'esnext-example',
 			title: 'ESNext Example',
 			description:
 				'Example block written with ESNext standard and JSX support – build step required.',
-			dashicon,
-			category,
-			author,
-			license,
-			licenseURI,
-			version,
 		},
-		outputFiles: [
-			'.editorconfig',
-			'.gitignore',
-			'editor.css',
-			'src/edit.js',
-			'src/index.js',
-			'src/save.js',
-			'$slug.php',
-			'style.css',
-			'readme.txt',
-		],
-		wpScriptsEnabled: true,
 	},
 };
 
-const getTemplate = ( templateName ) => {
-	if ( ! templates[ templateName ] ) {
+const getOutputTemplates = async ( name ) => {
+	const outputTemplatesPath = join( __dirname, 'templates', name );
+	const outputTemplatesFiles = await glob( '**/*.mustache', {
+		cwd: outputTemplatesPath,
+		dot: true,
+	} );
+	return fromPairs(
+		await Promise.all(
+			outputTemplatesFiles.map( async ( outputTemplateFile ) => {
+				const outputFile = outputTemplateFile.replace(
+					'.mustache',
+					''
+				);
+				const outputTemplate = await readFile(
```

Comment:
Probably outside the scope of what you're trying to do here, since as far as I know it's not fundamentally changing anything that doesn't already exist, but: I wonder if we should be lazier here. Rather than eagerly loading every template for every template option, at least only loading the template files for what's relevant given the user input.

It could be a matter of the template string being a function to call when needed, or something more "clever" (not necessarily a good thing) like `Object.defineProperty` getters, or only `getOutputTemplates` for templates relevant by input.

The last might also be good to avoid `fast-glob` if we know we won't need the template files.

Reply:
> It could be a matter of the template string being a function to call when needed

To clarify, it only loads all templates for a block template selected when calling `npm init @wordpress/block`. By default, it will only grab files from `lib/templates/esnext` folder. At the moment, we use all loaded templates to scaffold the project for the block. 

> Rather than eagerly loading every template for every template option, at least only loading the template files for what's relevant given the user input.

Once we have some conditions that make some templates options, it will make sense to defer loading templates and using a technique like " template string being a function" 👍 

Did I miss something?

Reply:
> To clarify, it only loads all templates for a block template selected when calling `npm init @wordpress/block`. By default, it will only grab files from `lib/templates/esnext` folder. At the moment, we use all loaded templates to scaffold the project for the block.

That's good, and largely addresses the concern I had. It was a misinterpretation on my part.

I was thinking about it a little more later in the day yesterday, still misled by the prior concern, but technically still valid: Ideally we try to avoid holding file content in memory longer than necessary. It's something which Node streams excel at, though at a more simple level it could just be a matter of: Only ever having one file loaded in memory at a time, generate the output, move on to the next (rely on garbage collection).

It could have been more of a problem if my previous impression of the implementation was correct, which it's not. Now it may be overkill 😄

Reply:
> Only ever having one file loaded in memory at a time, generate the output, move on to the next (rely on garbage collection).

We had that before and I liked it more. Here I prematurely assumed that template downloaded from npm would be load into memory and removed instantaneously. Let's see how the final implementation looks like. If we could make it work with `npm install my-template --no-save` then we could list absolute paths to the individual file instead and that would solve both issues assuming that npm cleans up the `node_modules` folder when using `npm init` :)